src/integrations/prefect-kubernetes/prefect_kubernetes/worker.py

```diff
@@ -621,7 +646,134 @@ async def run(
+        self,
+        flow: "Flow[..., R]",
+        parameters: dict[str, Any] | None = None,
+        job_variables: dict[str, Any] | None = None,
+    ) -> "PrefectFlowRunFuture[R]":
+        """
+        EXPERIMENTAL: The interface for this method is subject to change.
+
+        Submits a flow to run in a Kubernetes job.
+
+        Args:
+            flow: The flow to submit
+            parameters: The parameters to pass to the flow
+
+        Returns:
+            A flow run object
+        """
+        if self._runs_task_group is None:
+            raise RuntimeError("Worker not properly initialized")
+        flow_run = await self._runs_task_group.start(
+            partial(
+                self._submit_adhoc_run,
+                flow=flow,
+                parameters=parameters,
+                job_variables=job_variables,
+            ),
+        )
+        return PrefectFlowRunFuture(flow_run_id=flow_run.id)
+
+    async def _submit_adhoc_run(
+        self,
+        flow: "Flow[..., R]",
+        parameters: dict[str, Any] | None = None,
+        job_variables: dict[str, Any] | None = None,
+        task_status: anyio.abc.TaskStatus["FlowRun"] | None = None,
+    ):
+        """
+        Submits a flow run to the Kubernetes worker.
+        """
+        from prefect._experimental.bundles import (
+            convert_step_to_command,
+            create_bundle_for_flow_run,
+        )
+
+        if TYPE_CHECKING:
+            assert self._client is not None
+            assert self._work_pool is not None
+        flow_run = await self._client.create_flow_run(
+            flow, parameters=parameters, state=Pending()
+        )
+        if task_status is not None:
+            task_status.started(flow_run)
+        # Avoid an API call to get the flow
+        api_flow = APIFlow(id=flow_run.flow_id, name=flow.name, labels={})
+        logger = self.get_flow_run_logger(flow_run)
+
+        # TODO: Migrate steps to their own attributes on work pool when hardening this design
+        upload_step = json.loads(
+            get_from_dict(
+                self._work_pool.base_job_template,
+                "variables.properties.env.default.PREFECT__BUNDLE_UPLOAD_STEP",
+                "{}",
+            )
+        )
+        execute_step = json.loads(
+            get_from_dict(
+                self._work_pool.base_job_template,
+                "variables.properties.env.default.PREFECT__BUNDLE_EXECUTE_STEP",
+                "{}",
+            )
+        )
+
+        upload_command = convert_step_to_command(upload_step, str(flow_run.id))
+        execute_command = convert_step_to_command(execute_step, str(flow_run.id))
+
+        job_variables = (job_variables or {}) | {"command": " ".join(execute_command)}
+
+        configuration = await self.job_configuration.from_template_and_values(
+            base_job_template=self._work_pool.base_job_template,
+            values=job_variables | {"command": " ".join(execute_command)},
```

Comment:
this is redundant with the above on L725

Reply:
Updated in [d18f090](https://github.com/PrefectHQ/prefect/pull/17218/commits/d18f090343cde29cf037a7b11146ff684d7f1051)